toolchain/parser/parser_handle_index_expression.cpp

```diff
@@ -19,6 +19,13 @@ auto ParserHandleIndexExpression(ParserContext& context) -> void {
 
 auto ParserHandleIndexExpressionFinish(ParserContext& context) -> void {
   auto state = context.PopState();
+  if (!context.PositionIs(TokenKind::CloseSquareBracket)) {
+    CARBON_DIAGNOSTIC(UnexpectedTokenInIndex, Error,
+                      "Unexpected token in index expression");
+    context.emitter().Emit(*context.position(), UnexpectedTokenInIndex);
+    context.ReturnErrorOnState();
+    context.SkipTo(*context.FindNextOf({TokenKind::CloseSquareBracket}));
```

Comment:
You can use `tokens.GetMatchedClosingToken` to jump straight to the close bracket, if you can find the open bracket token; see `ParserContext::ConsumeAndAddCloseParen`.

Hm, in fact, I wonder if we can generalize that to work for other kinds of brackets and use it directly here?

Reply:
How's this?

Reply:
Looks great :) Thank you!